File: go/syntax/syntax.go
// This is the syntax for a single-line comment.

/* This is the syntax for a multi-line comment. As with other languages, Go
 *  ignores the contents of comments.*/

/* All go programs are part of a package. The below delcares that this program
 *  is part of the main package. Note that all executable Go programs require a
 *	main package.*/
package main

/* Statements in Go can be concluded with either a semi-colon or a new line. In
 * 	the latter case, a semi-colon is implicitly added to the line even if it
 *	isn't shown in the source code.*/

// Imports the files in the fmt package for use.
import (
	"fmt"
)

/* The func keyword is used to declare a function. Go requires that all main
 *	packages include a main function.
 *
 * Note that Go does not allow the opening bracket to be placed at the start of
 *	a line.*/
func main() {
	// Calls the Println function from the fmt package:
	fmt.Println("Hello world!")

}

File: c/syntax/syntax.c
// This is a single-line comment. The C language ignores whitespace.

/* This is a multi-line comment. The contents of comments are ignored by the 
 *  compiler.*/

/* The # symbol indicates a preprocessor directive. Preprocessor directives are 
 *  processed before the compilation of the file's remaining source code. 
 *
 * The include preprocessor directive tells the compiler to include the
 *  specified file. Here the specified file is the stdio.h header file, which
 *  is part of the C Standard Library. The angle brackets surrounding the
 *  filename indicate that the file resides in a standard system directory
 *  rather than the local project directory.*/
#include <stdio.h>

/* A function is a block of executable code. The code contained within the 
 *  brackets is executed. Functions can return a value, but are not required to. 
 *  The The main function is a special function that serves as the entry point 
 *  for the program.*/
int main()
{
    /* The individual instructions of which a program is comprised are referred
     *  to as statements. In C, each statement must end with a semi-colon. The
     *  statement below calls the printf function located in the stdio.h file.
     *  The printf function outputs the value in parentheses to the console:*/
    printf("Hello World!\n");

    // Note that text constants must be surrounded by quotation marks.

    /* The main function returns an integer (whole number) value. Here the value
     *  being returned is 0, which is typically used by the operating system to 
     *  indicate that the program executed successfully.*/
    return 0;

}